src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/RazorFormattingPass.cs

```diff
@@ -165,21 +165,55 @@ private static void TryFormatSingleLineDirective(FormattingContext context, List
             // @attribute [Obsolete("old")]
             //
             // The CSharpCodeBlockSyntax covers everything from the end of "attribute" to the end of the line
-            if (node is CSharpCodeBlockSyntax content &&
-                node.Parent?.Parent is RazorDirectiveSyntax directive &&
-                directive.DirectiveDescriptor?.Kind == DirectiveKind.SingleLine)
+            SyntaxList<SyntaxNode>? children;
+            if (IsSingleLineDirective(node, out children) ||
```

Comment:
Out of curiosity, does the C# formatter not do this already? Or if they do are we throwing away the edits?

Reply:
The C# formatter would, but it doesn't get to see the whole line. I _think_ this is the bad assumption I had when I logged the issue I mentioned. The way this is parsed is as `<razor>@</razor><csharp>using</csharp><razor> </razor><csharp>System</csharp>` so the formatter doesn't help. `@attribute` is the exception, where the formatter gets the whole directive, which is why there is a special case for it in the C# formatter, but usings need a special case in the razor formatter.

If that makes sense?

Reply:
> If that makes sense?

Yes totallly does! And because the C# formatter doesn't see the whole line it bails right?

Reply:
Kinda. If our C# formatter were to have a go at this, it would ask Roslyn's C# formatter to format `using`, then `System`, and whilst the formatter itself would fix the whole line, the transition mapping wouldn't see that they've moved.